libsolidity/analysis/TypeChecker.cpp

```diff
@@ -193,6 +193,18 @@ TypePointers TypeChecker::typeCheckABIDecodeAndRetrieveReturnType(FunctionCall c
 					typeArgument->location(),
 					"Decoding type " + actualType->toString(false) + " not supported."
 				);
+
+			if (auto referenceType = dynamic_cast<ReferenceType const*>(actualType))
+			{
+				auto result = referenceType->validForLocation(referenceType->location());
```

Comment:
Can be inlined inside the `if`

Reply:
Don't see how - it is negated.

Reply:
What's wrong with
`if (!referenceType->validForLocation(referenceType->location()))`
?

Reply:
`result` is needed further down

Reply:
missed that :p